### PR TITLE
Add publish.md

### PR DESCRIPTION
**Pull request by pbernasconi**

### CODE REVIEW

File: PUBLISH.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+```bash
+make release-(patch|minor|major)
+```
+
+Run the following git command to ensure the new version tag is pushed to the
+`plaid-node` GitHub repository:
+
+```bash
+git push --follow-tags
```
Comment (skansal-plaid) [inline PUBLISH.md]: IIRC this was only required for plaid-go and not plaid-node. xyz takes care of pushing git tags and branches directly to master.

File: PUBLISH.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
+# Publish guide
```
Comment (skansal-plaid) [inline PUBLISH.md]: We should also call out the process to obtain permissions for:
1. npmjs plaid-node package
2. github permissions to push directly to the protected master branch (or re-evaluate this restriction)

Comment (skansal-plaid) [inline PUBLISH.md]: We should also call out updating the changelog. Additionally, we should also consider introducing the quick process of documenting released on github for client libraries through https://github.com/plaid/plaid-node/releases/new

File: PUBLISH.md (hunk 1 above)
Comment (pbernasconi) [reply PUBLISH.md]: ⚡️

File: PUBLISH.md (hunk 2 above)
Comment (pbernasconi) [reply PUBLISH.md]: I'm gonna leave this out of this public md and put it in the private doc

Comment (pbernasconi) [reply PUBLISH.md]: ⚡️

**Pull request merged**